scripts/other-yes-list/parse.ts

```diff
@@ -81,3 +82,28 @@ export function genCourseEntries(data) {
   }
   return courseEntries.filter(course => !Number.isNaN(course.catalogNbr));
 }
+
+export function getValidCourses(data) {
+  const courseEntries = genCourseEntries(data);
+  const approvedCourses = fullCoursesArray;
+
+  const validCourses: CourseWithId[] = [];
+  for (const courseWithId of approvedCourses) {
+    for (const course of courseEntries) {
+      if (
+        courseWithId.subject === course.subject &&
+        courseWithId.catalogNbr === course.catalogNbr.toString()
+      ) {
+        const x: CourseWithId = {
+          subject: course.subject,
+          catalogNbr: course.catalogNbr,
+          categories: course.categories,
+          crseId: courseWithId.crseId,
+        };
+        validCourses.push(x);
+        break;
```

Comment:
could there ever be more than one `course` such that `courseWithId.subject === course.subject && courseWithId.catalogNbr === course.catalogNbr.toString()` is true? because by breaking after you find a match, you are saying that you are only looking for the first occurrence where that `if` condition passes.